crates/fs/src/fs.rs

```diff
@@ -562,10 +565,16 @@ impl Fs for RealFs {
                     .paths
                     .iter()
                     .filter_map(|path| {
-                        path.starts_with(&root_path).then(|| PathEvent {
-                            path: path.clone(),
-                            kind,
-                        })
+                        if path.starts_with(&root_path)
+                            || (target_root_path.as_ref().is_ok()
```

Comment:
We don't want to use an `is_ok()` / `unwrap()` pair if we can help it. Let's rewrite this to use an `if let Some(_) = _` :)